apps/docs/src/index.data.ts

```diff
@@ -0,0 +1,249 @@
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { defineLoader } from "vitepress";
+import type { ComponentGridProps } from "./.vitepress/components/ComponentGrid.vue";
+import type { Tab } from "./.vitepress/components/TabGroup.vue";
+
+/**
+ * Build-time data for the home page (components, facts/numbers etc.)
+ */
+export type HomePageData = {
+  /** Total number of implemented onyx components. */
+  componentCount: number;
+  /** Total number of component variants/stories across all implemented components as documented with Storybook. */
+  variantCount: number;
+  /** Total number of merged pull requests on GitHub. */
+  mergedPRCount: number;
+  /** Total number of closed issues on GitHub. */
+  closedIssueCount: number;
+  /** Total number of GitHub contributors. */
+  contributorCount: number;
+  /** Timestamp when this data has been fetched. */
+  timestamp: string;
+  /** Total number of npm downloads for all onyx npm packages in the last month. */
+  downloads: number;
+  /** Number of npm packages inside the `packages` folder of this monorepo. */
+  packageCount: number;
+  /** Component tabs / sets / roadmap. */
+  componentTabs: (Tab & ComponentGridProps)[];
+};
+
+declare const data: HomePageData;
+export { data };
+
+/**
+ * Build-Time data loader to get a list of available languages
+ * @see https://vitepress.dev/guide/data-loading
+ */
+export default defineLoader({
+  watch: ["../../../packages/sit-onyx/src/components/*/*.stories.ts"],
+  async load(watchedFiles): Promise<HomePageData> {
+    const variantCount = watchedFiles.reduce((total, file) => {
+      const fileContent = fs.readFileSync(file, "utf-8");
+      // stories are defined with e.g. "export const Primary =" so we can
+      // check the occurrences of "export const" to get the number of available stories
+      return total + countWord(fileContent, "export const");
```

Comment:
are we sure that we won't `export const demoData = ` in the stories in the future?
else checking against `} satisfies Story;` may be more reliable?

Reply:
Good idea :)